### PR TITLE
KP-568 - Refactor to use Kaholo AWS Plugin

### PR DESCRIPTION
**Pull request by kdebowski-kaholo**

### CODE REVIEW

**Review by Kszemi — DISMISSED** [review]

File: config.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "imgUrl": "logo.png",
   "execProgram": "node",
   "main": "app.js",
-  "version": "2.0.4",
+  "version": "2.1.0",
```
Comment (arispen) [inline config.json]: please change the version to 3.0.0

File: config.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "imgUrl": "logo.png",
```
Comment (arispen) [inline config.json]: please add "viewName": "Amazon S3"

Comment (arispen) [inline config.json:4]: please add "viewName": "Amazon S3"

File: config.json (hunk 1 above)
Comment (kdebowski-kaholo) [reply config.json]: are you sure? This is from similar PR in EC2
image

File: package.json [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
+	"devDependencies": {
+		"eslint": "8.8.0",
+		"eslint-config-airbnb-base": "15.0.0",
+		"eslint-plugin-import": "2.25.4",
+		"jest": "27.5.0"
+	},
 	"dependencies": {
-		"aws-sdk": "^2.1030.0"
+		"aws-sdk": "2.1030.0",
+		"kaholo-aws-plugin": "github:Kaholo/aws-plugin",
```
Comment (graddaniel) [inline package.json]:
```suggestion
                "kaholo-aws-plugin": "github:Kaholo/aws-library#1.0.0",
```

File: helpers.js [hunk 4]
```diff
@@ -1,65 +1,194 @@
...
-        return resolve(data);
-    }
+function combineGrants(currentGrants, newGrants) {
+  if (currentGrants.length === 0) { return newGrants; }
+
+  const grantsAreEqual = (a, b) => a.Permission === b.Permission
+        && a.Grantee.Type === b.Grantee.Type
+        && getGranteeId(a) === getGranteeId(b);
+
+  const applicableNewGrants = newGrants.reduce(
```
Comment (graddaniel) [inline helpers.js]: Why not just `newGrants.filter(newGrant => currentGrants.every(currentGrant => !grantsAreEqual(newGrant, currentGrant))`
or an alternative version depending which case is more likely
`newGrants.filter(newGrant => !currentGrants.some(currentGrant => grantsAreEqual(newGrant, currentGrant))`

**Review by graddaniel — CHANGES_REQUESTED** [review]

File: helpers.js (hunk 4 above)
Comment (kdebowski-kaholo) [reply helpers.js]: I 100% agree, I overengineered this one.

**Review by graddaniel — DISMISSED** [review]

**Review by graddaniel — APPROVED** [review]

**Pull request merged**